### PR TITLE
SAM debug: enable debugging for dotnet 3.1 Lambdas

### PR DESCRIPTION
**Pull request by abrooksv**
## Screenshots (if appropriate)
Screen Shot 2020-10-02 at 10 13 48 AM

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/sam/debugger/awsSamDebugger.ts [hunk 1]
```diff
@@ -331,18 +331,6 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
             vscode.Uri.parse(templateInvoke.templatePath!!)
 
         let awsCredentials: Credentials | undefined
-
-        // TODO: Remove this line to enable dotnetcore3.1 debugging when it becomes available
-        if (runtime === 'dotnetcore3.1' && !config.noDebug) {
```
Comment (awschristou) [inline src/shared/sam/debugger/awsSamDebugger.ts:336]: In the past, we've gated functionality around the detected SAM CLI versions when we didn't raise the minimum supported version.

Comment (abrooksv) [reply src/shared/sam/debugger/awsSamDebugger.ts:336]: Do we have an easy way to gate this? Raising min for everyone seems a little aggressive

**Review by justinmk3 — APPROVED** [review]

File: src/shared/sam/debugger/awsSamDebugger.ts (hunk 1 above)
Comment (awschristou) [reply src/shared/sam/debugger/awsSamDebugger.ts:336]: I agree that we wouldn't want to raise the min version
#785 has an example of version check gating

Comment (abrooksv) [reply src/shared/sam/debugger/awsSamDebugger.ts:336]: What would you have it do? Fail?

Comment (bryceitoc9) [reply src/shared/sam/debugger/awsSamDebugger.ts:336]: Maybe behave the same way as this but update the message with the minver for dotnetcore3.1 debugging?

**Review by bryceitoc9 — APPROVED** [review]

**Review by awschristou — APPROVED** [review]

File: src/shared/sam/debugger/awsSamDebugger.ts (hunk 1 above)
Comment (abrooksv) [reply src/shared/sam/debugger/awsSamDebugger.ts:336]: Added

**Pull request merged**